src/playturn.cpp

```diff
@@ -156,7 +156,8 @@ turn_info::PROCESS_DATA_RESULT turn_info::process_network_data(const config& cfg
 		game_display::get_singleton()->get_chat_manager().remove_observer(observer_quit["name"]);
 	}
 	else if (cfg.child("leave_game")) {
-		throw ingame_wesnothd_error("");
+		const bool has_reason = cfg.child("leave_game").has_attribute("reason");
+		throw ingame_wesnothd_error(has_reason ? cfg.child("leave_game").get("reason")->str() : "");
```

Comment:
even if you check for has_attribute before, is still better to use ["reason"] over get("reason")

Reply:
Done.